@here/harp-vectortile-datasource/lib/OmvDebugLabelsTile.ts

```diff
@@ -121,109 +139,146 @@ export class OmvDebugLabelsTile extends Tile {
-                bluePointIndices.push(pointIndex - 2);
-                bluePointIndices.push(pointIndex - 1);
-
-                if (debugTextPoisFull) {
-                    const offsetXY = pointSize * 0.5;
-                    const label: string = `${textElement.text} [${pointLabelIndex}]`;
-
-                    const labelElement = new TextElement(
-                        ContextualArabicConverter.instance.convert(label),
-                        new THREE.Vector3(
-                            x + worldOffsetX + centerX + offsetXY,
-                            y + centerY + offsetXY,
-                            z + centerZ
-                        ),
-                        textRenderStyle,
-                        textLayoutStyle,
-                        PRIORITY_ALWAYS,
-                        0.0,
-                        0.0
-                    );
-                    labelElement.minZoomLevel = 0;
-                    labelElement.mayOverlap = true;
-                    labelElement.reserveSpace = false;
-                    labelElement.alwaysOnTop = true;
-                    labelElement.ignoreDistance = true;
-                    labelElement.priority = TextElement.HIGHEST_PRIORITY;
+                const isLineMarker = textElement.type === TextElementType.LineMarker;
+                const geometry = isLineMarker ? orangePoints : bluePoints;
+                const pointSize = pointScale * 5;
+
+                const addLabel = (x: number, y: number, z: number) => {
+                    addPoint(x, y, z, pointSize, geometry);
+
+                    if (debugTextPoisFull || debugLineMarkers) {
+                        const offsetXY = pointSize * 0.5;
+                        const label: string = `${textElement.text} [${pointLabelIndex}]`;
+
+                        const labelElement = new TextElement(
+                            ContextualArabicConverter.instance.convert(label),
+                            new THREE.Vector3(
+                                x + worldOffsetX + centerX + offsetXY,
+                                y + centerY + offsetXY,
+                                z + centerZ
+                            ),
+                            textRenderStyle,
+                            textLayoutStyle,
+                            PRIORITY_ALWAYS,
+                            0.0,
+                            0.0
+                        );
+                        labelElement.minZoomLevel = 0;
+                        labelElement.mayOverlap = true;
+                        labelElement.reserveSpace = false;
+                        labelElement.alwaysOnTop = true;
+                        labelElement.ignoreDistance = true;
+                        labelElement.priority = TextElement.HIGHEST_PRIORITY;
+
+                        (labelElement as any)._isDebug = true;
+
+                        addedTextElements.push(labelElement);
+                    }
```

Comment:
consider also moving this function out unless it needs too many extra parameters.

Reply:
It would need 11 extra parameters... I think it should stay as a local function.